realm/realm-jni/src/io_realm_internal_tableview.cpp

```diff
@@ -34,7 +34,12 @@ inline bool view_valid_and_in_sync(JNIEnv* env, jlong nativeViewPtr) {
             ThrowException(env, TableInvalid, "The Realm has been closed and is no longer accessible.");
             return false;
         }
-        TV(nativeViewPtr)->sync_if_needed();
+        if (!TV(nativeViewPtr)->is_in_sync() && TV(nativeViewPtr)->depends_on_deleted_linklist()) {
```

Comment:
If a table view doesn't have any links list, will `depends_on_delete_linklist()` return true unconditionally? @rrrlasse, can you us a hint?